packages/components/src/card/card/hook.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useContextSystem } from '../../ui/context';
+import { useSurface } from '../../ui/surface';
+import * as styles from '../styles';
+
+/**
+ * @param {import('../../ui/context').PolymorphicComponentProps<import('../types').Props, 'div'>} props
+ */
+function useDeprecatedProps( { elevation, isElevated, ...otherProps } ) {
+	let computedElevation = elevation;
+
+	if ( isElevated ) {
+		deprecated( 'Card isElevated prop', {
+			since: '5.8',
+			alternative: 'elevation"',
+		} );
+		computedElevation ??= 2;
+	}
+
+	return {
+		...otherProps,
+		isElevated,
+		elevation: computedElevation,
+	};
+}
+
+/**
+ * @param {import('../../ui/context').PolymorphicComponentProps<import('../types').Props, 'div'>} props
+ */
+export function useCard( props ) {
+	const {
+		className,
+		elevation = 2,
+		isBorderless = false,
+		isRounded = true,
+		size = 'medium',
+		...otherProps
+	} = useContextSystem( useDeprecatedProps( props ), 'Card' );
+
+	const classes = useMemo( () => {
+		return cx(
+			styles.Card,
+			isBorderless && styles.boxShadowless,
```

Comment:
Isn't it a bit weird that "borderless" card actually means with no `box-shadow`? (see `styles.js` file)

What's the point of adding box-shadow if we're using `<Elevation />` too? We could simply say that, if `isBorderless=true`, then `elevation` is forced to a value of `0`